oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/query/async/ElasticAsyncQueryException.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.jackrabbit.oak.plugins.index.elastic.query.async;
+
+import co.elastic.clients.elasticsearch._types.ElasticsearchException;
+
+public class ElasticAsyncQueryException extends RuntimeException {
+
+    private final String esQuery;
+
+    public ElasticAsyncQueryException(Throwable cause, String esQuery) {
+        super(cause);
+
+        this.esQuery = esQuery;
+    }
+
+    @Override
+    public String getMessage() {
+        StringBuilder sb = new StringBuilder();
+        sb.append(String.format("Error retrieving data for ES query:\n\t%s", esQuery));
+        if (getCause() instanceof ElasticsearchException) {
+            ElasticsearchException esException = (ElasticsearchException) getCause();
+            sb.append("\n:: Elasticsearch response status: ").append(esException.status()).append(", message: ").append(esException.getMessage());
+            esException.error().rootCause().forEach(ec -> sb.append("\n\t* caused by ").append(ec.type()).append(": ").append(ec.reason()));
```

Comment:
I wonder if we need to protect against error() being null, rootCause() returning null, and so on.

Reply:
I checked the Elastic source code and both methods cannot return null